server/__tests__/socket.test.ts

```
@@ -26,56 +26,279 @@ afterEach(() => {
-test('initSocket uses allowed origins in production', () => {
-  process.env.NODE_ENV = 'production';
-  process.env.ALLOWED_ORIGINS = 'https://a.com,https://b.com';
-  const httpServer = {} as unknown as import('http').Server;
-  initSocket(httpServer);
-  expect(MockServer).toHaveBeenCalledWith(httpServer, {
-    cors: { origin: ['https://a.com', 'https://b.com'] },
+describe('Edge cases and error scenarios', () => {
+  test('handles missing process.env gracefully', () => {
+    const originalEnv = process.env;
+    // @ts-expect-error - Testing edge case
+    delete process.env;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)